client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -69,6 +69,7 @@
    private Marshaller marshaller;
    protected TransportFactory transportFactory;
    private ExecutorService asyncExecutorService;
+   private ExecutorService parallelExecutorService;
```

Comment:
Instead of having two executor services in the client, I'd prefer deprecating the ```asyncExecutorService``` and create a new config ```executorService``` that will handle both async and parallel operations. This way the client will have a single executor.